homeassistant/components/lcn/helpers.py

```diff
@@ -1,18 +1,191 @@
 """Helpers for LCN component."""
+import logging
 import re
 
+import pypck
 import voluptuous as vol
 
-from homeassistant.const import CONF_NAME
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_DEVICES,
+    CONF_DOMAIN,
+    CONF_ENTITIES,
+    CONF_HOST,
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_UNIQUE_ID,
+    CONF_USERNAME,
+)
+
+from .const import (
+    CONF_ADDRESS_ID,
+    CONF_CONNECTIONS,
+    CONF_DIM_MODE,
+    CONF_DOMAIN_DATA,
+    CONF_HARDWARE_SERIAL,
+    CONF_HARDWARE_TYPE,
+    CONF_IS_GROUP,
+    CONF_RESOURCE,
+    CONF_SEGMENT_ID,
+    CONF_SK_NUM_TRIES,
+    CONF_SOFTWARE_SERIAL,
+    CONF_UNIQUE_DEVICE_ID,
+    CONNECTION,
+    DEFAULT_NAME,
+    DOMAIN,
+)
 
-from .const import DEFAULT_NAME
+_LOGGER = logging.getLogger(__name__)
 
 # Regex for address validation
 PATTERN_ADDRESS = re.compile(
     "^((?P<conn_id>\\w+)\\.)?s?(?P<seg_id>\\d+)\\.(?P<type>m|g)?(?P<id>\\d+)$"
 )
 
 
+DOMAIN_LOOKUP = {
+    "binary_sensors": "binary_sensor",
+    "climates": "climate",
+    "covers": "cover",
+    "lights": "light",
+    "scenes": "scene",
+    "sensors": "sensor",
+    "switches": "switch",
+}
+
+
+def get_device_config(unique_device_id, config_entry):
+    """Return the device configuration for given unique_device_id from ConfigEntry."""
+    for device_config in config_entry.data[CONF_DEVICES]:
+        if device_config[CONF_UNIQUE_ID] == unique_device_id:
+            return device_config
+    return None
+
+
+def get_device_address(device_config):
+    """Return a tuple with address information."""
+    return (
+        device_config[CONF_SEGMENT_ID],
+        device_config[CONF_ADDRESS_ID],
+        device_config[CONF_IS_GROUP],
+    )
+
+
+def get_device_connection(hass, unique_device_id, config_entry):
+    """Return a lcn device_connection."""
+    device_config = get_device_config(unique_device_id, config_entry)
+    if device_config:
+        host_connection = hass.data[DOMAIN][config_entry.entry_id][CONNECTION]
+        addr = pypck.lcn_addr.LcnAddr(*get_device_address(device_config))
+        device_connection = host_connection.get_address_conn(addr)
+        return device_connection
+    return None
```

Comment:
It looks like most of the callers expect device_connection to be not-none (e.g. all the entities).  Perhaps fail sooner than later?

Reply:
I think there are still cases where misconfigurations will result in a hard to diagnose problem, so consider this one.

Reply:
I was able to remove the get_device_config method at this stage. The get_device_connection will always return a connection (at least a virtual one) from the api.